### ISSUE TITLE
[Bug]: Sequential analysis 'arrow' annotations are misleading

### ISSUE THREAD

**Issue by DIV-on-github** [jasp-stats/jasp-issues#1605]
### JASP Version

0.15

### JASP Module

jaspFrequencies

### What analysis are you seeing the problem on?

Bayesian Binomial Test

### What OS are you seeing the problem on?

Windows 10

### Bug Description

Sequential analysis charts are very useful in understanding a Bayesian analysis.
However, the 'arrow' annotations are misleading.  
When a non-informative/neutral prior is chosen, then the arrows correctly indicate what the evidence tells us, e.g. that there's moderate evidence in favour of H_0.  
However, when an informative prior is chosen, the arrows and chart are misleading.  What they actually show is the extent to which the evidence should make us change our prior perceptions (and in which direction).  However, the annotations don't indicate that;  instead they wrongly indicate that the evidence is directly interpretable as support for H_0 or H_1.  

### Expected Behaviour

The annotations should indicate that the plot represents the extent to which the evidence should make us change our prior perceptions (and in which direction).  
The annotations should probably include the word "**change**".  

### Steps to Reproduce

1. Open JASP.
2. Run the Hungry Zombies sample data.
3. In the second analysis, change the prior parameter "a" to 1000.  

Even though the evidence remains strongly in favour of H_0, BF_0- is merely 1.012, and the chart wrongly indicates that nett evidence for H_0 is almost non-existent (i.e. almost completely balanced with evidence for H_1).  

![image](https://user-images.githubusercontent.com/66234612/152913512-c1202ea7-6bb4-422a-baa2-48141f04362e.png)

—DIV

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if any.
- [X] I have included a JASP file that causes the crash/bug, if any.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by Kucharssim** [comment]
Dear @DIV-on-github,

thank you for the suggestion.

> Even though the evidence remains strongly in favour of H_0, BF_0- is merely 1.012, and the chart wrongly indicates that nett evidence for H_0 is almost non-existent

I would say that in this case the evidence is not strongly in favour of H_0, as the BF_0- = 1.012 indicates, because the evidence is relative depending on the models being compared. 

As this is more of a conceptual discussion: @EJWagenmakers, what do you think about this? I have no problem with changing the terminology or clarifying the plots in some way, but I was of the impression that we use the word "evidence" consistently throughout JASP, so I would not like to introduce a schism.

**Comment by DIV-on-github** [comment]
Thank-you for your comments, Kucharssim.  

From my point of view, I didn't feel it was an issue with the word "evidence" _per se_, but more so with H_0 and H_1.  
To me the evidence regarding H_0 indicates the 'overall' (posterior) probability that H_0 is correct, given the evidence.  
But instead the sequential graph seems to be presenting something (loosely) like the probability that H_0 is _more correct_ than what was _estimated _prior to viewing the evidence_, given the evidence.  It doesn't seem like an intuitive match to the annotations on that graph, to me.  
On the other hand, it seems that the annotations would be correct **if** H_0 and H_1 were 'conditional' hypotheses, _defined relative to the prior_.  

—DIV

P.S.  Frankly it _could_ be my misunderstanding, or else my misinterpretation.  
If it's my my misunderstanding, then no urgent action is required, but perhaps the documentation can be reviewed to address it.  
If it's my misinterpretation, then the GUI should still be reviewed to try to avoid such misinterpretation among users.

**Comment by DIV-on-github** [comment]
One more example to illustrate my query.  
If the prior is highly weighted to a proportion close to zero, then the posterior probability of H_0 becomes really small.
Yet the Sequential Analysis now reports that the "Evidence for H_0" is extreme.  

![image](https://user-images.githubusercontent.com/66234612/153112740-62353d95-9e90-4391-b62f-8f2d05d94eff.png)

I still feel that this should be something like "**_Additional_ evidence for H_0 arising from the observations**".
The intended implication of "_additional_" is that the prior acts as the 'baseline'.  

—DIV

**Comment by DIV-on-github** [comment]
Sorry, one more thought on this. 
As you know very well, Bayesian analysis can indeed be applied sequentially.  
Therefore what I have been referring to above as a "prior" could also be based on **prior _evidence_**.  
—DIV

**Comment by EJWagenmakers** [comment]
I am not sure I understand. H0 ("all zombies are hungry") is compared to H1, which can be defined in different ways. If you define H1 as "almost all zombies are hungry", then 12 hungry zombies constitute very little evidence -- that is, the data necessitate only a small adjustment of the prior odds. On the other hand, if you define H1 as "almost no zombies are hungry", then 12 hungry zombies will constitute compelling evidence against H1 (and, consequently, in favor of H0). 
Basically, evidence is a relative concept and depends on the relative predictive performance of two models for the observed data. This predictive performance is determined in part by the prior distribution.
Cheers,
E.J.

**Comment by EJWagenmakers** [comment]
....you argue that the evidence in favor of H0 can be extreme but that its posterior probability is very low. We do not provide the posterior model probabilities though. What you are probably looking at is the posterior distribution for the parameter under H1 at theta = 1. That is not the same as the posterior probability for H0. What does happen is that the posterior density at theta=1 (under H1) is much higher than it was before the data came in -- the ratio equals the Bayes factor, and this quantifies the extent to which the data drive a change in our beliefs.

**Comment by DIV-on-github** [comment]
> H0 ("all zombies are hungry") is compared to H1, which can be defined in different ways. 

Well, I suppose that's true, but the example indicates simply "For all tests, the alternative hypothesis specifies that the proportion is less than 1."  
I took that as 
H_1: theta < 1.
which is exactly what's indicated in the GUI under "Alt. Hypothesis" too (see screenshots above).  

> If you define H1 as "almost all zombies are hungry", then 12 hungry zombies constitute very little evidence -- that is, the data necessitate only a small adjustment of the prior odds. On the other hand, if you define H1 as "almost no zombies are hungry", then 12 hungry zombies will constitute compelling evidence against H1 (and, consequently, in favor of H0).

Yes, I agree with all of that.  
However, I do not see where such detailed H1 models would be specified in JASP.  Per your description, the model defined in H1 seems to be able to be expressed independently of the specification of the prior probability distribution.  
If so, then what model for H1 is being used in the sequential plot?

> What you are probably looking at is the posterior distribution for the parameter under H1 at theta = 1. 

Yes, I was.  My wording/interpretation was not precise.  Sorry about that.  

> That is not the same as the posterior probability for H0. 

I take your point.  

> What does happen is that the posterior density at theta=1 (under H1) is much higher than it was before the data came in -- the ratio equals the Bayes factor, and this quantifies the extent to which **the data drive a _change_ in our beliefs**.  [emphasis added —DIV]

I accept your explanation, but I still feel that some indication of "change" or "new" or "additional" would be helpful in presenting the annotations on the sequential plot.  (Maybe I still need to think about it some more.)  

—DIV

P.S.   _Thanks for your time, and I do appreciate the effort that has already gone into the development of JASP!_

**Comment by DIV-on-github** [comment]
I also came up with a further example that I thought might illuminate my concern.  

Note:  
Literally the JASP GUI uses the phrasing:  "Evidence for H_0".  I would say that this phrasing would be interpreted as "Evidence for H_0 being true".  

**Introduction to example**
Suppose that I have lived a sheltered life, and have not previously heard of zombies.  
However my parents and older siblings all have experience of zombies.  
Specifically — unbeknownst to me — my older sister has encountered 12 zombies (all hungry), and, independently, my older brother has coincidentally also encountered 12 zombies (all hungry).  [Let's suppose that we can safely assume that they were different zombies, in case that's a concern.]  

**Scenario 1**
One day my mother decides that I'm old enough now to be told to, "_Be careful if you see a zombie:  they eat humans._"
Based on this limited information about zombies, and my experience with other creatures that are not always hungry, I form the prior belief that encountered zombies are unlikely to all be hungry all of the time.  Perhaps I have a vague prior centred on theta = 0.5.  
Next my sister tells me about her experience.  Immediately I assess this additional information, and the resulting posterior probability distribution is much larger at theta = 1 than before, leading to a large Bayes factor, and JASP might tell me that this is (say) _strong evidence_ for H0:  theta = 1.  
Later my brother tells me about his experience.  Immediately I assess this new information (building upon the foregoing posterior distribution, which now becomes a new prior distribution), and the resulting new posterior probability distribution is only slightly larger at theta = 1 than before (after hearing from my sister), leading to a small Bayes factor, and JASP might tell me that this is (say) _anecdotal evidence_ for H0:  theta = 1.  
Conclusion:  my sister and brother both made equivalent (independent) observations, but my sister's account was rated as _strong evidence_ of H0 being true, and my brother's account as _weak evidence_ of H0 being true.  

**Scenario 2**
As before, but the sequence is altered.  First my mother raises a general warning, but then I hear about my brother's experience before finally hearing about my sister's experience.  

Conclusion:  my sister and brother both made equivalent (independent) observations, but my sister's account was rated as _weak evidence_ of H0 being true, and my brother's account as _strong evidence_ of H0 being true.  _I.e._ the opposite of the ratings in Scenario 1.  

**Discussion of example**
Personally I find it is very strange, and even wrong, to say that the account I hear second provides weaker evidence than the first account in regard to the **"truth of H0"**.  
More correctly it is something that should be reflecting _how much we change our mind_ about the "truth of H0".  
Both accounts (from my sister and my brother) are equally strong evidence (in favour of H0 being true), it's just that one makes me change my mind more than the other.  

**Proposal**
Maybe it should be written
**"Strong evidence _to strengthen one's belief in_ H0"** =  **"Strong evidence _to weaken one's belief in_ H1"**
**"Anecdotal evidence _to strengthen one's belief in_ H0"** =  **"Anecdotal evidence _to weaken one's belief in_ H1"**
**"Anecdotal evidence _to strengthen one's belief in_ H1"** = **"Anecdotal evidence _to weaken one's belief in_ H0"**
**"Strong evidence _to strengthen one's belief in_ H1"** =  **"Strong evidence _to weaken one's belief in_ H0"**
_et cetera_.
Except that it is a bit too wordy.  

In summary, 
* I don't have any concern about the use of the word "evidence";
* optionally it could be considered to clarify that it is "_new_ evidence";  
* emphasise that the (new) evidence's strength or weakness relates to how much it is changing our beliefs, not how likely the hypothesis is to be true.  

—DIV

**Comment by Kucharssim** [comment]
> Well, I suppose that's true, but the example indicates simply "For all tests, the alternative hypothesis specifies that the proportion is less than 1."
I took that as H_1: theta < 1. which is exactly what's indicated in the GUI under "Alt. Hypothesis" too (see screenshots above).


> However, I do not see where such detailed H1 models would be specified in JASP. Per your description, the model defined in H1 seems to be able to be expressed independently of the specification of the prior probability distribution.
If so, then what model for H1 is being used in the sequential plot?

Specification of the alternative hypothesis is done both by the selection under "Alt. Hypoyhesis" (which defines one-sided or two sided alternatives), and the parameters of the Beta priors, which define the shape of the prior under H_1. I agree though that we can be more explicit in the help file, which is quite vague about the Beta prior:

![Screenshot 2022-02-10 at 10 12 20](https://user-images.githubusercontent.com/7093546/153375019-fad8ba72-429d-4b38-92e3-5c07989e7a30.png)

We can also improve the footnote to say that the alternative hypothesis is also defined by the beta priors. The input panel could also be improved so that the Beta prior is under the "Alt. Hypothesis" heading.

> Even so, I would anticipate here that typically if the density of the posterior distribution at theta = 1 increases, then the posterior probability of H0 (being theta=1) also increases.

That is true, but remember that JASP does not show you the posterior probability of H_0, just the Bayes factor. Also, in the first screenshot you posted, the Bayes factor *is in favour* of H_0, and the posterior density at theta = 1 is larger than the prior, but only ever so slightly (by a factor or 1.012), because the prior distribution under H_1 that you set (Beta(1000, 1)) is making extremely similar predictions to H_0. So there is not much in the data to distinguish the two.

> Literally the JASP GUI uses the phrasing: "Evidence for H_0".
I would say that this phrasing would be interpreted as "Evidence for H_0 being true".
Grammatically I cannot see any way to make it mean something like "Evidence for H_0 well, actually, not really H_0 itself, but rather the change of our belief in H_0".

This is a fair point, a point that many people might be having misconceptions about. I do think though that we are consistent in pointing out that the evidence is always relative, e.g., BF_01 constitutes the evidence in favor of the specified null compared to the specified alternative hypothesis. See for example the help files:

![Screenshot 2022-02-10 at 10 23 28](https://user-images.githubusercontent.com/7093546/153376946-418b3cf3-441d-465a-b69f-14c3fac8e73e.png)

Perhaps we can be more explicit about it in the output itself (e.g., by displaying informative footnotes), although we simply cannot explain everything in the output explicitly, because the output would become cluttered. We are also not displaying footnotes explaining correct interpretation of p-values, despite people having misconceptions about them. We have to draw a line somewhere. For me, the line is around whether the concept is general enough to be understood out of the context of JASP, or whether it's something that JASP does uniquely and needs to be therefore clear about. The nature of Bayes factors, as well as p-values, are general enough concepts with clear explanations in many books and materials that I am not convinced that we ought to educate users on it, but this is not a very strong opinion of mine, more like a slight preference.

> Personally I find it is very strange, and even wrong, to say that the account I hear second provides weaker evidence than the first account in regard to the "truth of H0".
More correctly it is something that should be reflecting how much we change our mind about the "truth of H0".
Both accounts (from my sister and my brother) are equally strong evidence (in favour of H0 being true), it's just that one makes me change my mind more than the other.

Perhaps this is the cornerstone of this debate, because I would say the sentence:

> Both accounts (from my sister and my brother) are equally strong evidence (in favour of H0 being true), it's just that one makes me change my mind more than the other.

is not correct: Evidence *is* what changes your mind. Your brother and your sister happened to collect equivalent **data**, but that data provided different evidence because your hypotheses under consideration updated based on whoever's account you heard first. Evidence is always relative to the hypotheses you are comparing; it does not depend on the **data** alone. Naturally, then, equivalent data provide different evidence when the models under comparison are different. From this perspective, your sentence would be:

"Both accounts provided different evidence **because** one made me to change my mind more than the other, despite them both coming to me with equivalent observations."

I hope this makes it a bit clearer.

As much as I enjoy this discussion, I think we should move towards pragmatic conclusions. Based on this debate, I identified the following points we can do better in the analysis:

1. Be clearer in the help files about the meaning of the beta priors.
2. Reorganize the input panel to put the beta priors under the "Alt. hypothesis" heading.
3. Improve the footnote to also display what kind of beta prior is used on the alternative hypothesis.
4. (I am not conviced of this point, but not opposed to discussing it further): Be more clear in the **output** that Bayes factors are relative and do not point to "truthfulness" of hypotheses.


> P.S. Thanks for your time, and I do appreciate the effort that has already gone into the development of JASP!

No worries, it's great to receive feedback!

**Comment by EJWagenmakers** [comment]
Also note that more elaborate information is provided in the LearnBayes module -> Binomial Test. The help file could refer to that functionality.

**Comment by DIV-on-github** [comment]
Thanks, Kucharssim, for many constructive comments.  

> Perhaps we can be more explicit about it in the output itself (e.g., by displaying informative footnotes), although we simply cannot explain everything in the output explicitly, because the output would become cluttered. [...] We have to draw a line somewhere. 

I totally agree with this.  Hence it is difficult to come up with an improvement on "Evidence for H0" that remains _concise_.  
But I think "**New evidence towards H0**" could be a candidate:  it's not super-long, "towards" suggests movement (change), and personally I'm keen on "new" (see above).  

Another option, to avoid cluttering the display, is to clarify key phrases in the GUI through the use of _tooltips_, which are only visible when the user's mouse hovers over the relevant element.  

> Evidence is what changes your mind.

I bow to your expertise :-)
Maybe I had a more 'legalistic' interpretation.  (Imagine a courtroom scene:  "We submit exhibit A as evidence of my client's innocence.")
Or a wider philosophical view.  I can imagine having a moderately informative prior that subsequently is reinforced by _copious_ new data (n=10000, say), which I suppose you'd describe as "anecdotal evidence" (towards H0) because it wouldn't cause me to change my view.  If I next received _a small amount_ of contradictory data, I suppose you'd describe it as "anecdotal evidence" too (away from H0) because it wouldn't cause me to change my view much either.  But retrospectively I can say that if the first (large) batch of data hadn't been received, then the second (small) batch of data would have had a greater influence on my thinking — perhaps even causing me to prefer H1 over H0.  So, retrospectively, I can say that the first (large) batch of data _was_ actually highly impactful, because it largely 'inoculated' me against the (likely spurious) substantial change in opinion that would otherwise have been engendered by the second (small) batch of data.  

—DIV

**Comment by EJWagenmakers** [comment]
All of these intuitions about evidence are either captured by Bayes rule. or they are wrong :-)

You say: "Maybe I had a more 'legalistic' interpretation. (Imagine a courtroom scene: "We submit exhibit A as evidence of my client's innocence.")"
In this case, exhibit A is supposed to change the mind of the jury, in that exhibit A increases the probability of innocence compared to what it was before. The opposite would be preposterous ("We submit exhibit A as evidence of my client's innocence. It is now more likely that he is guilty"). So, evidence is the extent to which the data casuse a change in beliefs about the hypotheses under consideration. 

I personally think that adding "new" to evidence may be confusing. If this evidence is new, where is the old evidence? And given that we provided the data, isn't the evidence automatically what is new, namely what the data have just changed about our beliefs?  

E.J.

**Comment by DIV-on-github** [comment]
Thanks for the critique, E.J.  I am happy to learn from you.  :-)

Still with that in mind, I question your argument above.
> In this case, exhibit A is supposed to change the mind of the jury, in that exhibit A increases the probability of innocence compared to what it was before. [...] So, *evidence is the extent to which the data casuse a change in beliefs about the hypotheses under consideration*.  [emphasis added]

This scenario may not have been a perfect justification for my view, but I'm not (yet) convinced that it justifies your view either.  

Let's tweak the scenario a little.  
Now suppose the lawyer says to the jury, "We have already discussed exhibit A.  We now submit exhibit B into evidence.  This evidence proves conclusively that my client was not to blame.  After seeing this evidence you will surely find my client innocent!"

Notice that the lawyer:
* is making claims about the truth of a 'hypothesis';  
* is making predictions about the posterior views of the jury;  
* has no knowledge of the prior views of the jury;  and
* therefore could not assess whether any changes of opinion were occurring, and especially they couldn't know the size of any changes in opinion.  

Not being a lawyer myself, I imagine that what happens in a case such as the foregoing is that they present as much 'compelling evidence' as they can, in the hope that ultimately all jurors will vote in their favour — irrespective of their prior views.  

I can also imagine this conversation among members of such a legal team: 
Lawyer A - "What exhibits should we submit first?  Exhibit A could raise some 'reasonable doubt', but I think exhibit B is the most persuasive."
Lawyer B - "I agree.  We should present exhibit A first:  even though it's not as conclusive as exhibit B, it'll help to 'soften up' the jury, and it'd almost be enough on its own.  (It should get us about 90% of the way there.)  We'll save exhibit B until later, to 'seal the deal', and guarantee a verdict in our client's favour.  (And remember, it is human nature to focus on the most recent information.)"

-----

Separate from that, there are other legal proceedings, in certain jurisdictions, in which legal practitioners present evidence not with the motive of swaying opinion in a pre-determined direction, but rather purely to approach the truth of a matter.  
I have no expert knowledge of such things, but see perhaps https://en.wikipedia.org/wiki/Amicus_curiae , https://en.wikipedia.org/wiki/Royal_commission , and https://www.lawyersweekly.com.au/sme-law/24742-the-role-of-counsel-assisting-in-commissions-of-inquiry .  In the latter:
"_It is the duty of counsel assisting to establish the facts of the matter – this duty extends to eliciting evidence that support or contradict an issue that a commissioner would consider and make a factual finding._"

-----

Anyway, I think this is veering a little off topic.  The relevance, of course, is whether the terminology used in JASP would be intuitively understood outside of the 'in crowd'.  

—DIV

P.S.  I cannot imagine a lawyer standing up in court and saying to the jurors, "If you harbour any suspicions about my client, you will find this exhibit to be strongly persuasive evidence of their innocence.  (On the other hand, _if you currently think my client is innocent, then you will find my present argument to be weak_.)"!
